### ISSUE TITLE
Should aria-required be supported on radio buttons instead of (or in addition to) radio groups?

### ISSUE THREAD

**Issue by joanmarie** [w3c/aria#2669]
While doing some work in Chromium related to ARIA, I discovered that Chromium was treating `aria-required` as a global attribute. When I changed that to be compliant with the spec, a test related to reporting `aria-required` on radio buttons failed.

Doing some digging, I discovered:

* The test I broke had landed as part of [this commit](https://chromium-review.googlesource.com/c/chromium/src/+/648550) whose commit message states manually testing with JAWS, NVDA, and VoiceOver was performed. This makes me wonder if those screen readers are supporting (perhaps even expecting?) `aria-required` to be supported on radio buttons. (Orca checks the radio button first and if the state is absent, checks the group.)
* Chromium inherited the code in  [`AXNodeObject:IsRequired()`](https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/renderer/modules/accessibility/ax_node_object.cc;l=3386-3396;drc=cb9eddbf97baa142cb7fb6c551c52fd26982695c;bpv=0;bpt=0) from [WebKit](https://github.com/WebKit/WebKit/blob/7e871b506eaba52c7d74519aeb0047d702101331/Source/WebCore/accessibility/AccessibilityNodeObject.cpp#L1176-L1189). 
*  Unlike Chromium, WebKit checks to see if `required` is a supported property before exposing it. BUT if you look at the list of roles in [`AXCoreObject::supportsRequiredAttribute()`](https://github.com/WebKit/WebKit/blob/7e871b506eaba52c7d74519aeb0047d702101331/Source/WebCore/accessibility/AXCoreObject.cpp#L948-L973) you'll see that radio button is there along with radio group.
* Looking at [Firefox](https://github.com/mozilla-firefox/firefox/blob/a7d872e9d28072c2dd6d688d1b6c0493fcf7e94c/accessible/base/ARIAMap.cpp#L1425-L1437), it appears that `aria-required` is among the non-global ARIA states Gecko treats as global. There's even a code comment indicating `aria-required` is not global and pointing to this [bug](https://bugzilla.mozilla.org/show_bug.cgi?id=553117) that doesn't appear to be a high priority.

I personally don't have any opinion. But the spec and the implementations are not in sync.

**Comment by yatil** [comment]
Technically, I think when you have radio buttons, one button should always be selected by default. So the group will always have a value, and aria-required would not be… required. That said, at some point people started to use radio buttons, which have no default selection, which does not make sense to me, as you cannot unset a radio group anyway. So there is no way to go back to the initial value (apart from abandoning the form).

I wished the spec and maybe browsers could make it more clear that radio button groups without an initial selection are a bad practice?

**Comment by SaraSoueidan** [comment]
Opinion: I think it doesn't make much sense to mark a single or multiple radio buttons in a group as required, knowing that only one of them—any one of them—can be selected. 

I agree with @yatil that there must always be at least one selected in a group of radio buttons. Enforcing that somehow makes more sense than setting `aria-required` on individual buttons within a group. 

After all, the radio buttons get treated as one tab stop, which means the group itself is treated as one item. So it makes sense that the `aria-required` attribute be set on the group.
